Sixteenhomework

ssaltovsk1 opened the pull request:
HomeWork 16 - Написание тестов

Вопросы
- корректный вывод System.out , System.err
Попробовать flush() - не работает
- test03 - чтение из потока - правильно?

ssaltovsk1 committed:
HomeWork 06 - Написание тестов

Вопросы
- каждый тест в своем пакете, со всеми классами. Даже если мы их не меняем или не используем во время теста. Насколько правильно? Наверное, нет. Если проект большой, много копий
- как тестировать то, что в принципе не предусмотренно приложением? Сейчас это ничья.
т.е. мы должны представить, что вывод при ничье отличается от выигрыша и в качестве эталона, сказать, что строка должна быть не такой, как обычно?

Пофиксить
- считывание с консоли. Не Scanner, а I/O streams
- вывод провалов тестов сразу в тексте теста, а не внизу
- буква И (кодировка в проекте)

ssaltovsk1 committed:
HomeWork 04 - Работа со счетами клиентов

ssaltovsk1 committed:
HomeWork 16 - Написание тестов

Вопросы
- корректны вывод System.out , System.err
Попробовать flush() - не работает
- test03 - чтение из потока - правильно?

warlife3 commented on src/main/java/ru/otus/homeworks/hw06/test/test01/GameTest01.java:
```diff
@@ -0,0 +1,25 @@
+
+    private final Dice dice = new DiceTestImplTest01();
+
+    public void gameTest01() {
```
тут не зачем писать тесты на тестовые классы
имеет смысл писать тесты используя эти классы

warlife3 commented on src/main/java/ru/otus/homeworks/hw06/test/test02/GameTest02.java:
```diff
@@ -0,0 +1,28 @@
+
+public class GameTest02 {
+
+    private final DiceTest02 dice = new DiceImplTest02();
```
тут так же нужно проверять метод DiceImpl но не DiceImplTest02

warlife3 commented on src/main/java/ru/otus/homeworks/hw06/test/test03/Game.java:
```diff
@@ -0,0 +1,24 @@
+import ru.otus.homeworks.hw06.test.test03.GameWinnerPrinterTest03;
+import ru.otus.homeworks.hw06.test.Player;
+
+public class Game {
```
класс гем переписывать нельзя

warlife3 commented on src/main/java/ru/otus/homeworks/hw06/test/test03/GameWinnerConsolePrinterTest03.java:
```diff
@@ -0,0 +1,20 @@
+    public void printWinner(Player winner) {
+
+        System.out.printf("Победитель: %s%n", winner.getName());
+        // Так тест проходит:
```
тут ты проверяешь вывод текста но не логику игры
так же используется один и тот же игрок то есть всегда побеждает игорь
получается что он сам с собой играет

ssaltovsk1 committed:
HomeWork 16 - Написание тестов

- переписан метод GameWinnerConsolePrinterTest04
- вопрос, почему нужно два разных игрока, чтобы соревновались. А не один?
- как задавать expected String, если два разных варианта

warlife3 commented on src/main/java/ru/otus/homeworks/hw06/test/test04/GameWinnerConsolePrinterTest04.java:
```diff
@@ -0,0 +1,19 @@
+    @Override
+    public void printWinner(Player winner) {
+
+        if (winner.getName().equals(GameTest04.getIgor().getName())) {
```
дучше всего проверять по ссылке в данном случае
